webapp/views.py

```diff
@@ -198,13 +199,14 @@ def _get_resources(self):
                     resource_filter=self.RESOURCE_FILTER,
                 )
                 posts = get_json_feed_content(api_url, limit=3)
-                posts_length = len(posts)
-                if 'posts' not in feed_items.keys():
-                    feed_items['posts'] = {}
-
-                feed_items['posts'][group_name] = {}
-                feed_items['posts'][group_name]['posts'] = posts
-                feed_items['posts'][group_name]['group_name'] = group['name']
+                if posts:
+                    posts_length = len(posts)
```

Comment:
This doesn't seem to be used so can be removed.

Reply:
Its passed to the `_generate_pagination_queries` function below.

Reply:
Line 100 checks the length - what happens if posts_length is not defined? You should maybe define it at the top of the function? Not sure how python handles things like that.

Reply:
It creates the var if not defined. I will set it at the top though.